components/script/dom/htmlvideoelement.rs

```diff
@@ -82,4 +213,149 @@ impl HTMLVideoElementMethods for HTMLVideoElement {
         }
         self.video_height.get()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-video-poster
+    make_getter!(Poster, "poster");
+
+    // https://html.spec.whatwg.org/multipage/#dom-video-poster
+    make_setter!(SetPoster, "poster");
+}
+
+impl VirtualMethods for HTMLVideoElement {
+    fn super_type(&self) -> Option<&dyn VirtualMethods> {
+        Some(self.upcast::<HTMLMediaElement>() as &dyn VirtualMethods)
+    }
+
+    fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
+        self.super_type().unwrap().attribute_mutated(attr, mutation);
+
+        if mutation.new_value(attr).is_none() {
+            return;
+        }
+
+        match attr.local_name() {
+            &local_name!("poster") => {
+                self.fetch_poster_frame(self.Poster().to_string().as_str());
```

Comment:
We can obtain this from the AttributeMutation instead when new_value is Some, since AttrValue derefs to &str.